### ISSUE TITLE
Generator doesn't work, generates .js files

### ISSUE THREAD

**Issue by erichonkanen** [kimroen/ember-cli-coffeescript#86]
Hello,

I am using `"ember-cli-coffeescript": ">=0.10.0",` in my ember-cli project (0.13.1)

Running generator command such as `ember generate model foo` will generate app/models/foo.js

Is there something I am missing? How can I generate files in coffeescript?

**Comment by kimroen** [comment]
Hi!

That should work just fine, so that's strange.

**Comment by kimroen** [comment]
I'll close this, but please reopen if you're still having issues with it.

**Comment by andrewhavens** [comment]
Please reopen this. I am also running into this issue.

I am working on migrating my `ember-rails` app to `ember-cli-rails`, so I just created my first Ember CLI app at the root of my Rails app directory:

    $ ember new frontend --skip-git
    $ ember install ember-cli-rails-addon

Next I installed the `ember-cli-coffeescript` add on and tried to generate some things:
 
    $ ember install ember-cli-coffeescript
    $ ember generate adapter application
    installing adapter
      create app/adapters/application.js
    installing adapter-test
      create tests/unit/adapters/application-test.js
    $ ember generate model comment body:string
    installing model
      create app/models/comment.js
    installing model-test
      create tests/unit/models/comment-test.js

I even tried manually setting the configuration in `config/environment.js`:

```js
module.exports = function(environment) {
  var ENV = {
    modulePrefix: 'frontend',
    environment: environment,
    baseURL: '/',
    locationType: 'auto',
    EmberENV: {
      FEATURES: {
        // Here you can enable experimental features on an ember canary build
        // e.g. 'with-controller': true
      }
    },
    coffeeOptions: {
      blueprints: true
    },

    APP: {
      // Here you can pass flags/options to your application instance
      // when it is created
    }
  };
// ...
```

The package is listed in `package.json`:

```js
  "devDependencies": {
    "broccoli-asset-rev": "^2.4.2",
    "ember-ajax": "0.7.1",
    "ember-cli": "2.4.3",
    "ember-cli-app-version": "^1.0.0",
    "ember-cli-babel": "^5.1.6",
    "ember-cli-coffeescript": "1.13.2",
// ...
```

What am I missing?

**Comment by andrewhavens** [comment]
Interesting...when I tried to generate a route, it looks like it started to create some `.coffee` files, but then it failed:
```
$ ember g route feed
installing route
  create app/routes/feed.coffee
  create app/templates/feed.hbs
ENOENT: no such file or directory, open '/Users/andrew/myapp/frontend/app/router.coffee'
Error: ENOENT: no such file or directory, open '/Users/andrew/myapp/frontend/app/router.coffee'
  at Error (native)
  at Object.fs.openSync (fs.js:584:18)
  at Object.fs.readFileSync (fs.js:431:33)
  at writeRoute (/Users/andrew/myapp/frontend/node_modules/ember-cli-coffeescript/blueprints/route/index.js:70:19)
  at Class.updateRouter (/Users/andrew/myapp/frontend/node_modules/ember-cli-coffeescript/blueprints/route/index.js:49:5)
  at Class.module.exports.afterInstall (/Users/andrew/myapp/frontend/node_modules/ember-cli-coffeescript/blueprints/route/index.js:32:18)
  at lib$rsvp$$internal$$tryCatch (/Users/andrew/myapp/frontend/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:1036:16)
  at lib$rsvp$$internal$$invokeCallback (/Users/andrew/myapp/frontend/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:1048:17)
  at lib$rsvp$$internal$$publish (/Users/andrew/myapp/frontend/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:1019:11)
  at lib$rsvp$asap$$flush (/Users/andrew/myapp/frontend/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:1198:9)
  at nextTickCallbackWith0Args (node.js:452:9)
  at process._tickCallback (node.js:381:13)
```

**Comment by kimroen** [comment]
So sorry for not getting back to you on this. Reopening now.

It sounds like some other addon is providing blueprints, and those are being prioritized higher than the ones in this project. I checked to see if ember-cli-rails or ember-cli-rails addon has any blueprints, and it doesn't look like it so I doubt they are related.

From your examples though, it looks like the ones that don't work are ember-data related, so it could be that [the blueprints that were moved to the ember-data addon](https://github.com/emberjs/data/tree/master/blueprints) fairly recently are overriding.

> Interesting...when I tried to generate a route, it looks like it started to create some .coffee files, but then it failed

This error is not related to the problem - it's just that you don't have a `router.coffee` file, but one was expected. [I have an issue to detect this and delegate to the built-in blueprints](https://github.com/kimroen/ember-cli-coffeescript/issues/99)

Fixing the problem with blueprints being overridden is a matter of [adding `ember-data` to the list of addons this addon should run before](https://github.com/kimroen/ember-cli-coffeescript/blob/d317f2e4fa6a154332a9ce07a56035d39c0ad0dd/package.json#L67-L69), as far as I can tell. Hopefully that will do the trick.

Sorry again that I completely missed your comments in this issue.
